### PR TITLE
Fixed Battle class, load game meta from server, heartbeat, applied item checking, multi-transfer and map incense pokemon

### PR DESCRIPTION
**Pull request by Gegy**
This PR contains:

- A fixed Battle class
- Load game meta from server
- Heartbeat so that the API updates the map instead of the API user
- Applied item checking, to check if incense or lucky eggs are active
- Multi-Pokemon-Transfer
- Map incense pokemon
- Path speed changing
- Improved login flow
- Fixed pokemon from eggs not existing when listener is called

### CODE REVIEW

**Comment by Gegy** [comment]
This closes #826, #825, #816, #804, #778, #534, #557, #258 and #252.

File: library/src/main/java/com/pokegoapi/api/PokemonGo.java [hunk 1]
```diff
@@ -192,14 +203,40 @@ public void login(CredentialProvider credentialProvider)
 		playerProfile = new PlayerProfile(this);
 		playerProfile.updateProfile();
 
+		try {
```
Comment (PokeAlert) [inline library/src/main/java/com/pokegoapi/api/PokemonGo.java]: Nope.. I'd advise you to redo a MITM you will see whats up during a captcha login
(Login sequence goes all the way through and the common request exlude the captcha message)

File: library/src/main/java/com/pokegoapi/api/PokemonGo.java [hunk 2]
```diff
@@ -192,14 +203,40 @@ public void login(CredentialProvider credentialProvider)
...
 
 		this.loggingIn = false;
+
+		heartbeat.start();
 	}
 
 	private void initialize() throws RemoteServerException, CaptchaActiveException, LoginFailedException {
-		fireRequestBlock(new ServerRequest(RequestType.DOWNLOAD_REMOTE_CONFIG_VERSION,
-				CommonRequests.getDownloadRemoteConfigVersionMessageRequest()));
+		ServerRequest request = new ServerRequest(RequestType.DOWNLOAD_REMOTE_CONFIG_VERSION,
```
Comment (PokeAlert) [inline library/src/main/java/com/pokegoapi/api/PokemonGo.java]: There is a lot of missing request in here (initialize())

Comment (Gegy) [reply library/src/main/java/com/pokegoapi/api/PokemonGo.java]: You mean requests such as GET_PLAYER? Those are sent when the PlayerProfile is created, iirc.

Comment (PokeAlert) [reply library/src/main/java/com/pokegoapi/api/PokemonGo.java]: I dont remember all of them on top of my head but there ASSET_DIGEST, PLAYER_PROFILE, LEVEL_UP_REWARDS and some others

**Comment by vagprogrammer** [comment]
@gegy1000  Excellent job, you are a master. It will be avaiable soon? How can I acces to this changes?

**Comment by Gegy** [comment]
@vagprogrammer You can either build from my fork, or wait until this is merged.

**Comment by Gegy** [comment]
@PokeAlert, does the login flow look correct to you now, after the last commit? Here's a log of the requests sent during login: https://gist.github.com/2cf2237ae95c61312eabbdfc4d8f6345

**Pull request merged**